### DISCUSSION TITLE
Connect/Disconnect handling of scales

### DISCUSSION THREAD

**Discussion by obiwan007** [obiwan007/despresso#70] [Ideas]
Currently the app is trying to find a scale in the moment the de1 is heated up coming from sleep.
Additionally you can just click the connect button.

How should that be handled better?

Unfortunately I can not wake up a disconnected/switched off scale - so this is no option.

- I could add another "search for BLE device" if the de1 is waking up
- I could disconnect any scale if the de1 wil go to sleep - the scale then probably goes to sleep. I am not aware of any command where you could send a scale directly to sleepmode (at least not for Lunar or Scale2).
- I could immediately stop the de1 if you started the espresso without a connected scale (if you configured this).

**Comment by christophedc0** [comment]
I'm just thinking out loud here.

I think DSX continuously scans for a scale to be connected, when you launch the app and that works fine for me.

So what if..
You remember the BLE address of the previously connected scale, then search for x time to connect with that scale, when launching (or waking up) the Despresso app.
In combination with:
> I could immediately stop the de1 if you started the espresso without a connected scale (if you configured this).
